derive/src/decode.rs

```diff
@@ -46,7 +46,7 @@ pub fn quote(data: &Data, type_name: &Ident, input: &TokenStream) -> TokenStream
 
 			let recurse = data_variants().enumerate().map(|(i, v)| {
 				let name = &v.ident;
-				let index = utils::index(v, i);
+				let index = utils::get_variant_index(v, i);
```

Comment:
as you are cleaning up, isn't `variant_index` more rust idiomatic than `get_variant_index` ?
Anyway i don't mind either.

Reply:
Good pt, changing.